utility/mm2_update-master-directory-list

```diff
@@ -423,6 +455,84 @@ def short_filelist(files):
         return files
 
 
+def sync_category_directory(
+        session, config, category, relativeDName, ctime, files):
+
+    logger.debug("  sync_category_directory %s - %s" % (
+        category, relativeDName))
+
+    topdir = category.topdir.name
+
+    created = False
+    set_ctime = False
+    relativeDName = relativeDName.replace(topdir, '')
+    if relativeDName.startswith('/'):
+        relativeDName = relativeDName[1:]
+    if relativeDName in category.directory_cache:
+        d = category.directory_cache[relativeDName]
+        if d['ctime'] != ctime:
+            set_ctime = True
+            logger.info('ctime changed: ctime in database: %d - ctime on disk: %d' % (d['ctime'], ctime))
+        D = mirrormanager2.lib.get_directory_by_id(session, d.id)
+    else:
+        if relativeDName == u'':
+            D = category.topdir
+        else:
+            # Can't find the new directory, just add it
+            dname = os.path.join(topdir, relativeDName)
+            D = Directory(name=dname, ctime=ctime)
+            logger.debug(u'Created Directory(%s, ctime=%s)' % (dname, ctime))
+            created = True
+        # Add this category to the directory
+        D.categories.append(category)
+        session.add(D)
+        # And flush so that we can already start using it
+        session.flush()
+        # Refresh the cache
+        category.directory_cache[relativeDName] = D
+
+    if set_ctime:
+        D.ctime = ctime
+        logger.info('Setting ctime to %d' % ctime)
+
+    if set_ctime or created:
+
+        # stating changed files to see if they are readable
+        s = None
+        try:
+            s = os.stat(os.path.join(
+                config.get('UMDL_PREFIX', ''),
+                topdir,
+                relativeDName))
+        except OSError:
+            logger.debug("Hmm, stat()ing %s failed. Theoretically this cannot happen." % relativeDName)
+
+        if s:
+            mode = s.st_mode
+            readable = not not (mode & stat.S_IRWXO & (stat.S_IROTH|stat.S_IXOTH))
```

Comment:
This definitely could use a comment. I think what it does is check to see whether or not the file is readable or executable by others?

Applying ``not`` twice is also very odd. Maybe you just want to cast it to a boolean with ``bool``?